components/sup-protocol/src/lib.rs

```diff
@@ -119,19 +122,29 @@ where
                 ),
             ));
         }
-        File::open(&secret_key_path)
-            .and_then(|mut f| f.read_to_string(out))
-            .map_err(move |e| {
-                net::err(
+        match File::open(&secret_key_path) {
+            Ok(file) => match BufReader::new(file).lines().nth(0) {
+                Some(Ok(line)) => {
+                    *out = line.to_string();
+                    Ok(true)
+                }
+                _ => Err(net::err(
                     ErrCode::Io,
                     format!(
-                        "IoError while reading or writing ctl secret, {}, {}",
+                        "Could not find secret in ctl secret file, {}",
                         secret_key_path.display(),
-                        e
                     ),
-                )
-            })?;
```

Comment:
I think this change is sufficient to fix the error. I'd prefer to keep the change as small as possible.
```rust
        *out = out.trim_end().into();
```

Reply:
Nice recommendation. When I first read the `trim_end()` docs I didn't realize it would also handle CRLF that is typical on Windows.  That's great!